Utils/Functions/Net/GithubReleases/script.js

```diff
@@ -1,29 +1,41 @@
 const Downloader = include("utils.functions.net.download");
-const {createTempDir, remove, cat} = include("utils.functions.filesystem.files");
+const Resource = include("utils.functions.net.resource");
+const { createTempDir, remove, cat } = include("utils.functions.filesystem.files");
 
 /**
- * Get an array of tag corresponding to the releases in a git repository
+ * Fetches an array of tag corresponding to the releases in a git repository
  *
  * @param {string} repositoryOwner owner of the repository
  * @param {string} repositoryName name of the repository
  * @param {wizard} wizard the wizard
- * @returns {string[]} list of version tag in the repository
+ * @returns {object[]} list of version objects in the repository
  */
-module.getGithubReleases = function getGithubReleases(repositoryOwner, repositoryName, wizard) {
+module.fetchGithubReleases = function fetchGithubReleases(repositoryOwner, repositoryName, wizard) {
     const tmpDir = createTempDir();
 
     const releasesFile = new Downloader()
         .wizard(wizard)
-        .url(
-            `https://api.github.com/repos/${repositoryOwner}/${repositoryName}/releases`
-        )
+        .url(`https://api.github.com/repos/${repositoryOwner}/${repositoryName}/releases`)
         .message(tr("Fetching versions list..."))
         .to(tmpDir + "/releases.json")
         .get();
 
-    const versions = JSON.parse(cat(releasesFile)).map(version => version.tag_name);
+    const versions = JSON.parse(cat(releasesFile));
 
     remove(tmpDir);
 
     return versions;
 };
+
+module.extractGithubReleaseStrings = function extractGithubReleaseStrings(versions) {
+    return versions.map(version => version.name);
+}
+
+module.downloadGithubRelease = function downloadGithubRelease(version, wizard) {
+    return new Resource()
+        .wizard(wizard)
+        .url(version.tarball_url)
+        // TODO: generate a more "random" name to prevent file conflicts
+        .name(`${version.name}.tar.xz`)
```

Comment:
I don't like the file name. Ideally I think we should somehow generate a random filename here. @plata any ideas?

Reply:
Why not make the interface similar to `Downloader`?

Reply:
You mean change it to a class?

Reply:
Yes, maybe `GitHubDownloader` and then similar methods to `Downloader`. Would that make sense to you?

Reply:
Done